system/lib/libc/musl/src/thread/pthread_mutex_timedlock.c

```diff
@@ -79,6 +86,9 @@ int __pthread_mutex_timedlock(pthread_mutex_t *restrict m, const struct timespec
 		int own = r & 0x3fffffff;
 		if (!own && (!r || (type&4)))
 			continue;
+#if defined(__EMSCRIPTEN__) && !defined(NDEBUG)
+		assert(own != __pthread_self()->tid && "pthread mutex deadlock detected");
+#endif
 		if ((type&3) == PTHREAD_MUTEX_ERRORCHECK
```

Comment:
Can we perhaps enable this `PTHREAD_MUTEX_ERRORCHECK` globally somehow, rather than add these three lines? (It wouldn't assert though, just return EDEADLK)